modules/ui_common.py

```diff
@@ -36,6 +36,29 @@ def plaintext_to_html(text, classname=None):
     return f"<p class='{classname}'>{content}</p>" if classname else f"<p>{content}</p>"
 
 
+def update_logfile(logfile_path, fields):
+    import csv
+
+    with open(logfile_path, "r", encoding="utf8", newline="") as file:
+        reader = csv.reader(file)
+        rows = list(reader)
+
+    # blank file: leave it as is
+    if not rows:
+        return
+
+    rows[0] = fields
+
+    # append new fields to each row as empty values
+    for row in rows[1:]:
+        while len(row) < len(fields):
+            row.append("")

```

Comment:
This is a bit nit picky on my part but this approach could be a bit more pythonic.
```diff
-    for row in rows[1:]:
-        while len(row) < len(fields):
-            row.append("")
+    rows = [row + ([""] * (len(fields) - len(row))) for row in rows[1:]]
```

Reply:
hmm I'll have to push back on this, while list comprehensions are usually my preference, this becomes nigh unreadable.

Reply:
> hmm I'll have to push back on this, while list comprehensions are usually my preference, this becomes nigh unreadable.

Yeah fair enough.